frontend/frontend.go

```diff
@@ -10,15 +10,19 @@ import (
 // response from a parsed request, and (2) asynchronously observe anything
 // after the response has been delivered to the client.
 type TrackerLogic interface {
-	// HandleAnnounce generates a response for an Announce.
-	HandleAnnounce(context.Context, *bittorrent.AnnounceRequest) (*bittorrent.AnnounceResponse, error)
+	// HandleAnnounce generates a response for an Announce. Returns the updated
```

Comment:
Ah, sorry about the trouble, but: We usually have at least a newline, typically even an empty line before the "Returns ...", like so:

```go
// HandleAnnounce generates a response for an Announce.
//
// Returns the updated context, the generated AnnounceResponse and no
// error on success; nil and error on failure.
```
(I don't know if that fits within the line-length limit, only have a browser at hand.)